config/src/main/java/org/keycloak/config/Option.java

```diff
@@ -0,0 +1,56 @@
+package org.keycloak.config;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class Option<T> {
+    private final Class<T> type;
+    private final String key;
+    private final OptionCategory category;
+    private final boolean hidden;
+    private final boolean build;
```

Comment:
not really specific to the runtime, but specific to our distribution configuration model.
It's important to propagate this information so that different runtimes can make use of it.